### PR TITLE
TF2.8: Clean up dockerfiles, update HVD test

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
- refactor TF2.8 to use multistage dockerfiles
- remove sm package from e3 image
- update HVD test with the latest version from HVD repo

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 1]
```diff
@@ -165,3 +174,65 @@ RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.7/license.txt -o
 RUN rm -rf /tmp/*
 
 CMD ["/usr/bin/tf_serving_entrypoint.sh"]
+
```
Comment (arjkesh) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:177]: @hsl89 - can the commented lines regarding TensorRT in this file be removed?

Comment (hsl89) [reply tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:177]: yes, they are from the previous release

**Review by Qingzi-Lan — DISMISSED** [review]

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 2]
```diff
@@ -165,3 +174,65 @@ RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.7/license.txt -o
...
+FROM e3 AS sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# Specify accept-bind-to-port LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+
+ARG PYTHON=python3.8
```
Comment (hsl89) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:202]: We built the TF wheel against py39. Even this PR is not about SageMaker release, it is still better to keep it in sync with the E3 image.

File: tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu [hunk 3]
```diff
@@ -165,3 +174,65 @@ RUN curl https://aws-dlc-licenses.s3.amazonaws.com/tensorflow-2.7/license.txt -o
...
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.8.9
+ARG TFS_SHORT_VERSION=2.7
```
Comment (hsl89) [inline tensorflow/inference/docker/2.8/py3/cu112/Dockerfile.gpu:205]: change this to 2.8

**Review by hsl89 — CHANGES_REQUESTED** [review]

**Review by hsl89 — APPROVED** [review]
LGTM

**Review by Qingzi-Lan — APPROVED** [review]

**Pull request merged**